### DISCUSSION TITLE
Use Target Markdown with `knitr::knit_expand()`?

### DISCUSSION THREAD

**Discussion by 33Vito** [ropensci/targets#647] [Ideas]
Hi folks, 

I have been using the Target Markdown recently. I have also used [knitr::knit_expand()](https://bookdown.org/yihui/rmarkdown-cookbook/option-code.html) before to render Rmarkdown based on templates with R expressions in {{}}. So this idea just comes up, what if I use the following R markdown template (saved as `Targets_Rmarkdown_Template.Rmd`)

```
# {{name}}

```{targets {{name}}}
tar_target({{name}}, {
  {{code}}
})
```

I can feed it with a bunch of R script name and the corresponding code via `knit_expand()`, then render the results as child R markdown.

```
```{r, echo=FALSE, results='asis'}
src = lapply(c("Step1", "Step2"), function(i) {
  knitr::knit_expand('Targets_Rmarkdown_Template.Rmd', 
                     name = i, 
                     code = readr::read_file(paste0(i,".R"))
                     )
})
res = knitr::knit_child(text = unlist(src), quiet = TRUE)
cat(res, sep = '\n')
```

This means I can turn a series of R script into a Target Markdown workflow, without changing anything, and the dependency in these R scripts will be picked up by the targets engine. I tried that and it works. **However**, in the real world, it is more likely that these R scripts are dependent based on their script or output names, e.g. `Step1.R`, `Step2.R`, `Step1_output.csv`, `Step2_output.csv`, etc. for them to be run in order, rather than sharing variable names in the R expression. So this approach would not work that well. 

Anyone thinks there is any merit doing this, compared with writing a normal Target Markdown which wraps the workflows scattered in those R scripts (e.g. wrap the R scripts into Rmarkdowns, then feed into the targets pipeline via `tar_render()`?

**Comment by wlandau** [comment]
That's an interesting way of doing things. Does it feel more convenient for you? 

I could see `knitr::knit_expand()` as a way to handle multi-project workflows, maybe even https://github.com/ropensci/targets/discussions/595. Seems more like something users would opt into on their own, rather than something I would build into `targets`. But it could become more of a patten if more people use it, and I would probably document it somewhere at that point.

**Reply by 33Vito** [reply]
It's just an idea. I will give it a go when I had a chance. I think it can be useful when I need to break a long R script into multiple small scripts to make it more maintainable in production, will keep you updated :)
